docs/generated/stablehlo_passes.md

```diff
@@ -338,6 +338,143 @@ Modules valid for shape refinement must have the following properties:
   * All calls to a single function resolve to the same argument shapes, and no
     recursive / co-recursive function calls are made.
 
+### `-stablehlo-wrap-in-composite`
+
+_Wraps a non-composite  StableHLO op in a composite op._
+
+Wraps StableHLO operations in `stablehlo.composite` operations.
+
+For instance, consider a simple StableHLO program:
+
+```mlir
+func.func @main(%arg0 : tensor<2xf32>, %arg1 : tensor<2xf32>) -> tensor<2xf32> {
+  %0 = stablehlo.add %arg0, %arg1 : tensor<2xf32>
+  return %0 : tensor<2xf32>
+}
+```
+
+Applying this pass to wrap `stablehlo.add` operations will result in the
+following program:
+
+```mlir
+func.func @main(%arg0: tensor<2xf32>, %arg1: tensor<2xf32>) -> tensor<2xf32> {
+  %0 = stablehlo.composite "stablehlo.add" %arg0, %arg1 {decomposition = @stablehlo.add.impl} : (tensor<2xf32>, tensor<2xf32>) -> tensor<2xf32>
+  return %0 : tensor<2xf32>
+}
+func.func private @stablehlo.add.impl(%arg0: tensor<2xf32>, %arg1: tensor<2xf32>) -> tensor<2xf32> {
+  %0 = stablehlo.add %arg0, %arg1 : tensor<2xf32>
+  return %0 : tensor<2xf32>
+}
+```
+
+Notes:
+
+  - The `name` attribute of the generated `stablehlo.composite` operation
+    will always be the same as the name of the original operation that was
+    wrapped (e.g., if you wrap a `stablehlo.add` operation, the composite
+    will also be named `"stablehlo.add"`).
+  - The private function that encapsulates the original operation
+    (referenced by the `decomposition` attribute of the
+    `stablehlo.composite` operation) will be named using the pattern
+    `<op_name>.impl[.N]`, where `<op_name>` is the name of the original
+    operation, and `N` is a unique integer identifier generated to prevent
+    naming conflicts within the module.
+
+This pass can be used in three distinct ways:
+
+**Mode 1: Command-line Usage**
+
+This mode is the simplest, using the `stablehlo-opt` utility with the
```

Comment:
Remove this paragraph, options are covered later (auto generated) and all passes work with opt